File: src/main/scala/org/xbery/artbeams/comments/admin/CommentForm.scala
package org.xbery.artbeams.comments.admin

import net.formio.validation.validators.{EmailValidator, NotEmptyValidator}
import net.formio.{Field, FormMapping, Forms}
import org.xbery.artbeams.comments.domain.EditedComment
import org.xbery.artbeams.common.form.ScalaForms

/**
  * Comment edit form.
  * @author Radek Beran
  */
class CommentForm {
  lazy val definition: FormMapping[EditedComment] = {
    Forms.basic(classOf[EditedComment], "comment")
      .field("id", Field.HIDDEN)
      .field("entityId", Field.HIDDEN)
      .field("comment", Field.TEXT)
      .field("userName", Field.TEXT)
      .field(Forms.field("email", Field.TEXT).validator(EmailValidator.getInstance()))
      .field(Forms.field("antispamQuestion", Field.HIDDEN).validator(NotEmptyValidator.getInstance()))
      .field(Forms.field("antispamAnswer", Field.TEXT).validator(NotEmptyValidator.getInstance()))
      .build(ScalaForms.CzConfig)
  }
}

File: src/main/kotlin/org/xbery/artbeams/comments/admin/CommentForm.kt
package org.xbery.artbeams.comments.admin

import net.formio.Field
import net.formio.FormMapping
import net.formio.Forms
import net.formio.validation.Validator
import net.formio.validation.validators.EmailValidator
import net.formio.validation.validators.NotEmptyValidator
import org.xbery.artbeams.comments.domain.EditedComment
import org.xbery.artbeams.common.form.FormUtils

/**
 * Comment edit form.
 * @author Radek Beran
 */
open class CommentForm {
    companion object {
        val definition: FormMapping<EditedComment> =
            Forms.basic(EditedComment::class.java, "comment")
                .field<String>("id", Field.HIDDEN)
                .field<String>("entityId", Field.HIDDEN)
                .field<String>("comment", Field.TEXT)
                .field<String>("userName", Field.TEXT)
                .field(Forms.field<String>("email", Field.TEXT).validator(EmailValidator.getInstance()))
                .field(Forms.field<Object>("antispamQuestion", Field.HIDDEN).validator(NotEmptyValidator.getInstance() as Validator<Object>))
                .field(Forms.field<Object>("antispamAnswer", Field.TEXT).validator(NotEmptyValidator.getInstance() as Validator<Object>))
                .build(FormUtils.CzConfig)
    }
}
